docs/source/examples/Widget Custom.ipynb

```diff
@@ -0,0 +1,785 @@
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### Rendering model contents"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "By replacing the string literal with a call to `model.get`, the view will now display the value of the back end upon display.  However, it will not update itself to a new value when the value changes.\n",
+    "\n",
+    "```typescript\n",
+    "export class EmailView extends DOMWidgetView {\n",
+    "  render() {\n",
+    "    this._emailInput = document.createElement('input');\n",
+    "    this._emailInput.type = 'email';\n",
+    "    this._emailInput.value = this.model.get('value');\n",
+    "    this._emailInput.disabled = this.model.get('disabled');\n",
+    "      \n",
+    "    this.el.appendChild(this._emailInput);\n",
+    "  }\n",
+    "\n",
+    "  private _emailInput: HTMLInputElement;\n",
+    "}\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### Dynamic updates"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To get the view to update itself dynamically, register a function to update the view's value when the model's `value` property changes.  This can be done using the `model.on` method.  The `on` method takes three parameters, an event name, callback handle, and callback context.   The Backbone event named `change` will fire whenever the model changes.  By appending `:value` to it, you tell Backbone to only listen to the change event of the `value` property (as seen below).\n",
+    "\n",
+    "```typescript\n",
+    "export class EmailView extends DOMWidgetView {\n",
+    "  render() {\n",
+    "    this._emailInput = document.createElement('input');\n",
+    "    this._emailInput.type = 'email';\n",
+    "    this._emailInput.value = this.model.get('value');\n",
+    "    this._emailInput.disabled = this.model.get('disabled');\n",
+    "\n",
+    "    this.el.appendChild(this._emailInput);\n",
+    "\n",
+    "    // Python -> JavaScript update\n",
+    "    this.model.on('change:value', this._onValueChanged, this);\n",
+    "    this.model.on('change:disabled', this._onDisabledChanged, this);\n",
+    "  }\n",
+    "\n",
+    "  private _onValueChanged() {\n",
+    "    this._emailInput.value = this.model.get('value');\n",
+    "  }\n",
+    "\n",
+    "  private _onDisabledChanged() {\n",
+    "    this._emailInput.disabled = this.model.get('disabled');\n",
+    "  }\n",
+    "\n",
+    "  private _emailInput: HTMLInputElement;\n",
+    "}\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This allows us to update the value from the Python kernel to the views. Now to get the value updated from the front-end to the Python kernel (when the input is not disabled) we can do it using the `model.set` method.\n",
```

Comment:
Need to also mention `model.save_changes`. I'm pretty sure the difference is that `set` will set the ts model and all thew views will be synced together, and then on `save_changes` the python side will updated.

Reply:
```suggestion
    "This allows us to update the value from the Python kernel to the views. Now to get the value updated from the front-end to the Python kernel (when the input is not disabled) we set the value on the frontend model using `model.set` and then sync the frontend model with the Python object using `model.save_changes`.\n",
```